backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Tests/Controllers/ResourceControllerTest.cs

```diff
@@ -64,6 +65,127 @@ public async Task GetMaskinportenResources_valid()
+            Assert.Equal(expectedResult.NumEntriesTotal, actualResources.NumEntriesTotal);
+            AssertionUtil.AssertCollections(expectedResult.PageList, actualResources.PageList, AssertionUtil.AssertResourceExternalEqual);
+        }
+
+        /// <summary>
+        /// Test case: PaginatedSearch with pagination and filters
+        /// Expected: PaginatedSearch returns a list of resources matching the filters in paginated form with language filtered for the authenticated users selected language
+        /// </summary>
+        [Fact]
+        public async Task GetSingleRightsSearch_filterSet_ReturnsMatch()
+        {
+            // Arrange
+            var token = PrincipalUtil.GetToken(1337, 501337, 2);
+            _client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+            int page = 1;
+            int numPerPage = 4;
+            string[] roFilters = { "narnia", "testdepartementet" };
+
+            List<ServiceResourceFE> allExpectedResources = TestDataUtil.GetSingleRightsResources().FindAll(r => roFilters.Contains(r.ResourceOwnerName.ToLower()));
+            PaginatedList<ServiceResourceFE> expectedResult = new PaginatedList<ServiceResourceFE>(allExpectedResources.GetRange(0, numPerPage), page, allExpectedResources.Count);
+
+            // Act
+            HttpResponseMessage response = await _client.GetAsync($"accessmanagement/api/v1/resources/paginatedSearch?NumPerPage={numPerPage}&Page={page}&ROFilters={roFilters[0]}&ROFilters={roFilters[1]}");
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            PaginatedList<ServiceResourceFE> actualResources = JsonSerializer.Deserialize<PaginatedList<ServiceResourceFE>>(await response.Content.ReadAsStringAsync(), options);
+            Assert.Equal(expectedResult.Page, actualResources.Page);
+            Assert.Equal(expectedResult.NumEntriesTotal, actualResources.NumEntriesTotal);
+            AssertionUtil.AssertCollections(expectedResult.PageList, actualResources.PageList, AssertionUtil.AssertResourceExternalEqual);
+        }
+
+        /// <summary>
+        /// Test case: PaginatedSearch with pagination and search string
+        /// Expected: PaginatedSearch returns a list of resources matching the search string in paginated form with language filtered for the authenticated users selected language
+        /// </summary>
+        [Fact]
+        public async Task GetSingleRightsSearch_searchStringSet_ReturnsMatch()
+        {
+            // Arrange
+            var token = PrincipalUtil.GetToken(1337, 501337, 2);
+            _client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+            int page = 1;
+            int numPerPage = 1;
+            string searchString = "klesskapet";
+
+            List<ServiceResourceFE> allExpectedResources = TestDataUtil.GetSingleRightsResources().FindAll(r => r.Title.ToLower().Contains(searchString));
```

Comment:
Why don't you reuse the methods you created in the RAP instead of duplicating the same thing rap.SearchInResourceList does on this line?

Reply:
Because then I wouldn't actually be testing whether the code ran as it should and errors may pass as expected results. If I check the code by using the same code, it would be sorta like checking if a nail is the same as a hammer by comparing it to a picture of that same nail I am currently expecting.